File: Testing/Stdlib Test Code/test_default_params.py
from dataclasses import dataclass, field

def __ul_sort(arr, n): arr[:n] = sorted(arr[:n])

class __ul_Exception(Exception):
    def __init__(self, type_name, message):
        self.type_name = type_name
        self.message = message
        super().__init__(message)


def greet(name, greeting='Hello'):
    return '{greeting}, {name}!'

def add(a, b=10):
    return (a + b)

def main():
    print(greet('World', 'Hello'))
    print(greet('World', 'Hi'))
    print(add(5, 10))
    print(add(5, 20))
    return 0


if __name__ == "__main__":
    main()

File: Testing/Stdlib Test Code/test_default_params.js

function greet(name, greeting = "Hello") {
    return "{greeting}, {name}!";
}

function add(a, b = 10) {
    return (a + b);
}

function main() {
    console.log(greet("World", "Hello"));
    console.log(greet("World", "Hi"));
    console.log(add(5, 10));
    console.log(add(5, 20));
    return 0;
}

function __ul_sort(arr, n) { const sub = arr.slice(0, n).sort((a, b) => a < b ? -1 : a > b ? 1 : 0); for (let i = 0; i < n; i++) arr[i] = sub[i]; }

main();
